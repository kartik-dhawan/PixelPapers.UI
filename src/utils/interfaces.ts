export interface ChildrenProps {
  children: React.ReactNode
}

export type ContentfulFields = {
  [key: string]: any
}

export interface ThumbnailItem {
  fields: {
    blogMetaTitle: string
    blogTravelLocation: string
    blogType: string
    blogMetaImage: any
  }
  sys: any
}
export interface SocialMediaLink {
  id: number
  label: string
  url: string
  icon: string
}

export interface TravelBlogPost {
  blogTitle?: string
  blogMetaImage?: any
  blogMetaTitle?: string
  blogTravelLocation?: string
  blogType?: string
  blogUrlSlug?: string
  blogInitialDescription?: any
  blogThemeColor?: string
}

export interface ContentfulTravelBlog {
  fields: TravelBlogPost
  sys: any
  metadata: any
}
