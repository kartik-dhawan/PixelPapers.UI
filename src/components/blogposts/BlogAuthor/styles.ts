export const styles = {
  blogAuthorAvatarSectionWrapper: {
    padding: "0rem 2rem",
    boxSizing: "border-box",
    width: "100%",
  },
  blogAuthorAvatarWrapper: {
    height: "72px",
    width: "72px",
    margin: "1rem",
    borderRadius: "50%",
    "& img": {
      borderRadius: "50%",
      objectFit: "cover",
    },
  },
  blogAuthorDividers: { flexGrow: 1, borderWidth: "1px" },
  blogAuthorUsernameLink: {
    color: "#222",
    fontSize: {
      xs: "26px",
      xl: "30px",
    },
    "& a": {
      color: "#222",
      textDecoration: "none",
    },
    fontWeight: 500,
  },
  blogAuthorBio: {
    maxWidth: "720px",
    textAlign: "center",
    fontSize: {
      xs: "18px",
      xl: "20px",
    },
    margin: "1rem 0rem 1rem",
    fontWeight: 300,
    color: "#555",
    padding: {
      xs: "0rem 1rem",
      sm: "0rem 2rem",
    },
    boxSizing: "border-box",
  },
  blogAuthorGetInTouchBtn: {
    backgroundColor: "#222",
    borderRadius: "200px",
    margin: "1rem 0rem",
    fontWeight: 600,
    textTransform: "capitalize",
    fontSize: {
      xs: "16px",
      xl: "18px",
    },
    padding: "8px 24px",
    color: "#d9d9d9",
    "&:hover": {
      backgroundColor: "#353535",
    },
  },
  // blogAuthor
}
