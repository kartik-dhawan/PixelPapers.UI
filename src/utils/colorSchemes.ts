export const GLOBAL_COLORS = {
  TEXT_PRIMARY_DARK: "#222",
  TEXT_PRIMARY_LIGHTER: "#666",
  TEXT_PRIMARY_LIGHT: "#d9d9d9",
  BACKGROUND_PRIMARY_LIGHT: "#fefefe",
  BACKGROUND_PRIMARY_DARK: "#111",
  COMMON_FOOTER_BACKGRUND_DARK: "#383b3e",
  BLOG_SUB_COMPONENT_BACKGROUND_LIGHTER: "#efefef",
}

export const PAGE_LEVEL_COLORS = {
  PAGE_404: {
    BACKGROUND_PRIMARY_LIGHT: "#226bb152",
    TEXT_PRIMARY_LIGHT: "#444",
    TEXT_PRIMARY_DARK: "#1c354c",
  },
  TRAVEL: {
    BACKGROUND_PRIMARY_LIGHT: "#fdeded",
    BACKGROUND_PRIMARY_DARK: "#864040",
    TEXT_PRIMARY_LIGHTER: "#444",
  },
}
