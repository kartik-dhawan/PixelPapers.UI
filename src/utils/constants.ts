export const CURATED_BY_LINK_DEFAULT_TEXT = "kartik-dhawan"

export const DEFAULT_EMAIL = "contact.kartikdhawan@gmail.com"

export const AVERAGE_TIME_TO_READ_A_BLOG /**In minutes */ = 7

export const META_DATA_FALLBACK = {
  GENERIC: {
    title: "Pixel Papers | Personally curated blogs.",
    description:
      "A blogging website with an extra touch of emotions & real experiences. Unlike mechanical blogs on huge blogging websites, here you will feel home.",
  },
  TRAVEL: {
    title: "Travel | Pixel Papers",
    description:
      "A blogging website with an extra touch of emotions & real experiences. Unlike mechanical blogs on huge blogging websites, here you will feel home.",
  },
}

export const ANONYMOUS_AUTHOR_DATA = {
  USERNAME: "Anonymous",
  BIO: `"There are people, who are better unknown" - Hozier`, // eslint-disable-line
}

export const PODCAST_IFRAME_LINKS_DEFAULT = {
  SPOTIFY:
    "https://open.spotify.com/embed/show/0zvlbjXYlI1HyiGLLvul7Z?utm_source=generator",
  YOUTUBE:
    "https://www.youtube.com/embed/videoseries?si=XAlNUSdw1KLiZ940&amp;list=PLuusiVNsh08-eBJ3r7YLAzDvzZGtuGHHc",
}
