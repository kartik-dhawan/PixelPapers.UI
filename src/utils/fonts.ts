import { DM_Sans, Anton, Libre_Barcode_39 } from "next/font/google"

export const dmSans = DM_Sans({
  weight: ["200", "300", "400", "500", "600", "700"],
  subsets: ["latin"],
  display: "swap",
  preload: true,
})

export const barcode = Libre_Barcode_39({
  weight: ["400"],
  subsets: ["latin"],
  display: "swap",
})

export const anton = Anton({
  weight: ["400"],
  subsets: ["latin"],
  display: "swap",
})
