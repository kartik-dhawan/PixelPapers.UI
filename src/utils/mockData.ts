export const TRAVEL_PAGE_THUMBNAIL_CARD_DATA = {
  blogTitle: "Top 10 Things to do on your next trip to Jibhi/Jalori Pass.",
  blogMetaImage: {
    metadata: {
      tags: [],
    },
    sys: {
      space: {
        sys: {
          type: "Link",
          linkType: "Space",
          id: "1v98waf66sfj",
        },
      },
      id: "6TVtXnjJIZ812Bxm6VBqz5",
      type: "Asset",
      createdAt: "2023-11-21T17:39:08.757Z",
      updatedAt: "2023-11-21T17:39:17.165Z",
      environment: {
        sys: {
          id: "master",
          type: "Link",
          linkType: "Environment",
        },
      },
      revision: 2,
      locale: "en-US",
    },
    fields: {
      title: "Jibhi_Jalori",
      description:
        "A picture of sunset from Raghupur fort close to jalori pass",
      file: {
        url: "//images.ctfassets.net/1v98waf66sfj/6TVtXnjJIZ812Bxm6VBqz5/5b89ce3dced278570a2a663453d69a65/IMG_3749.webp",
        details: {
          size: 71536,
          image: {
            width: 3024,
            height: 4032,
          },
        },
        fileName: "IMG_3749.webp",
        contentType: "image/webp",
      },
    },
  },
  blogMetaTitle: "Jibhi - Jalori",
  blogTravelLocation: "Himachal Pradesh",
  blogType: "Travel Itinerary",
}
