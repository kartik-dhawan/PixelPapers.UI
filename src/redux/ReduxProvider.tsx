"use client"

/**
 *
 * This component acts as a parent for client side components that require redux store
 * Wrap the component with this 'Redux Provider' to use state management on client side
 *
 * the updates made on the client side in the store will not be reflected in the server side components
 *
 * SERVER SIDE:
 * - Dispatching: store.dispatch(functionName(payload))
 * - Getting data from store: store.getState().sliaceName.stateName
 *
 * CLIENT SIDE:
 * - Dispatching: dispatch(functionName(payload)) [useDispatch()]
 * - Getting data from store: useSelector((state)=>state.slice.statename)
 *
 * Ref: https://www.youtube.com/watch?v=dRLjoT4r-jc
 */

import { Provider } from "react-redux"
import { store } from "./store"
import { ChildrenProps } from "@/utils/interfaces"

const ReduxProvider = ({ children }: ChildrenProps) => {
  return <Provider store={store}>{children}</Provider>
}

export default ReduxProvider
