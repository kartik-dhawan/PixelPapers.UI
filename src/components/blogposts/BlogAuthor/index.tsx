"use client"

import { dmSans } from "@/utils/fonts"
import { BlogAuthorData, TravelBlogPost } from "@/utils/interfaces"
import { Box, Button, Divider, Stack, SvgIcon } from "@mui/material"
import Image from "next/image"
import Link from "next/link"
import MailOutlineIcon from "@mui/icons-material/MailOutline"
import { useEffect, useState } from "react"
import { styles } from "./styles"
import { ANONYMOUS_AUTHOR_DATA } from "@/utils/constants"

interface BlogAuthorProps {
  currentBlog: TravelBlogPost
  content: any
}

const BlogAuthor = ({ currentBlog, content }: BlogAuthorProps) => {
  const initialAuthorState: BlogAuthorData = {
    bio: "",
    username: "",
    usernameLink: "",
    email: "",
  }
  const [author, setAuthor] = useState<BlogAuthorData>(initialAuthorState)
  const [authorImage, setAuthorImage] = useState<any>()

  useEffect(() => {
    if (currentBlog.blogAuthorData !== undefined) {
      // if data of third party author is available
      // set the data as it is, with a nullify condition

      // if the third party author wants to be anonymous, they need to send an empty json object
      const realAuthor = currentBlog.blogAuthorData
      setAuthor({
        username: realAuthor.username ?? ANONYMOUS_AUTHOR_DATA.USERNAME,
        usernameLink: realAuthor.usernameLink ?? "#!",
        bio: realAuthor.bio ?? ANONYMOUS_AUTHOR_DATA.BIO,
        email: realAuthor.email,
      })
    } else {
      // display it as my own blog if the blogAuthorFields are empty
      setAuthor(content.myAuthorProfileData)
    }
    console.log(currentBlog.blogAuthorData)
  }, [content, currentBlog])

  useEffect(() => {
    console.log(author)
  }, [author])

  useEffect(() => {
    if (currentBlog.authorImage && currentBlog.blogAuthorData) {
      // if we have both image & authorData for third party author
      // then display the image of that author
      setAuthorImage(currentBlog.authorImage)
    } else if (currentBlog.blogAuthorData && !currentBlog.authorImage) {
      // if we do not have the image of the third party author but have the authorData
      // then show a generic image
      setAuthorImage(undefined)
    } else {
      // in every other case, show my default image
      // making the blog my own
      setAuthorImage(content.myProfileImage)
    }
  }, [content, currentBlog])

  return (
    <Stack
      alignItems="center"
      className={"blogAuthorSectionWrapper " + dmSans.className}
      component="section"
      marginBottom="3rem"
    >
      <Box fontSize={{ xs: "14px", xl: "18px" }} margin="1rem">
        by
      </Box>
      <Stack
        direction="row"
        divider={
          <Box
            sx={{
              backgroundColor: currentBlog.blogThemeColor,
              position: "relative",
              ...styles.blogAuthorAvatarWrapper,
            }}
          >
            {authorImage && (
              <Image
                src={`https:${authorImage?.fields.file?.url}`}
                alt={
                  authorImage.fields.description ??
                  "Profile Avatar of the Author of the Blog"
                }
                fill
                sizes="(max-width: 768px) 50vw, (max-width: 1200px) 30vw, 15vw)"
              />
            )}
          </Box>
        }
        alignItems="center"
        sx={styles.blogAuthorAvatarSectionWrapper}
      >
        <Divider sx={styles.blogAuthorDividers} />
        <Divider sx={styles.blogAuthorDividers} />
      </Stack>
      <Box sx={styles.blogAuthorUsernameLink}>
        {author.usernameLink ? (
          <Link
            href={author.usernameLink}
            target="_blank"
            rel="noreferrer"
            aria-label="External link to the profile of the author of this blog."
          >
            {author.username}
          </Link>
        ) : (
          author.username
        )}
      </Box>
      <Box sx={styles.blogAuthorBio}>{author.bio ?? "Let me be unknown."}</Box>
      {author.email && (
        <Link
          href={`mailto:${author.email}`}
          target="_blank"
          rel="noreferrer"
          aria-label="Link to the email of the author of this blog."
        >
          <Button
            sx={styles.blogAuthorGetInTouchBtn}
            startIcon={
              <SvgIcon>
                <MailOutlineIcon />
              </SvgIcon>
            }
          >
            Get in touch
          </Button>
        </Link>
      )}
    </Stack>
  )
}

export default BlogAuthor
