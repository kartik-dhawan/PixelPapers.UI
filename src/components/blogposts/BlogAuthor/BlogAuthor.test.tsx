import ReduxProvider from "@/redux/ReduxProvider"
import "@testing-library/jest-dom"
import renderer from "react-test-renderer"
import { BLOG_DATA_MOCK } from "@/utils/mockData"
import BlogAuthor from "."

describe("Blog Author Section Component", () => {
  it("-- it should match the snapshot", () => {
    const tree = renderer.create(
      <ReduxProvider>
        <BlogAuthor currentBlog={BLOG_DATA_MOCK.fields} content={{}} />
      </ReduxProvider>,
    )

    expect(tree).toMatchSnapshot()
  })
})
