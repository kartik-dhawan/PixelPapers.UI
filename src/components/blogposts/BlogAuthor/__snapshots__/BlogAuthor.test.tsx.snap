// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog Author Section Component -- it should match the snapshot 1`] = `
<section
  className="MuiStack-root blogAuthorSectionWrapper className css-1bu8ci5-MuiStack-root"
>
  <div
    className="MuiBox-root css-1kpt9ie"
  >
    by
  </div>
  <div
    className="MuiStack-root css-1o9heaf-MuiStack-root"
  >
    <hr
      className="MuiDivider-root MuiDivider-fullWidth css-1ec78s4-MuiDivider-root"
    />
    <div
      className="MuiBox-root css-y5mt8b"
    />
    <hr
      className="MuiDivider-root MuiDivider-fullWidth css-1ec78s4-MuiDivider-root"
    />
  </div>
  <div
    className="MuiBox-root css-9qdcri"
  />
  <div
    className="MuiBox-root css-tt3dop"
  />
</section>
`;
